plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -432,38 +432,42 @@ func podName(pod *api.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
-// PodFitsResources calculates fit based on requested, rather than used resources
-func (r *ResourceFit) PodFitsResources(pod *api.Pod, existingPods []*api.Pod, node string) (bool, error) {
-	info, err := r.info.GetNodeInfo(node)
-	if err != nil {
-		return false, err
-	}
-
+func PodFitsResource2(pod *api.Pod, existingPods []*api.Pod, info *api.Node) (bool, error) {
+	nodeName := info.Name
 	allocatable := info.Status.Allocatable
-	if int64(len(existingPods))+1 > allocatable.Pods().Value() {
-		return false, newInsufficientResourceError(podCountResourceName, 1,
-			int64(len(existingPods)), allocatable.Pods().Value())
-	}
-
 	podRequest := getResourceRequest(pod)
 	if podRequest.milliCPU == 0 && podRequest.memory == 0 {
 		return true, nil
 	}
-
 	pods := append(existingPods, pod)
 	_, exceedingCPU, exceedingMemory := CheckPodsExceedingFreeResources(pods, allocatable)
 	if len(exceedingCPU) > 0 {
-		return false, newInsufficientResourceError(cpuResourceName, podRequest.milliCPU,
+		return false, NewInsufficientResourceError(cpuResourceName, podRequest.milliCPU,
 			getTotalResourceRequest(existingPods).milliCPU, allocatable.Cpu().MilliValue())
 	}
 	if len(exceedingMemory) > 0 {
-		return false, newInsufficientResourceError(memoryResoureceName, podRequest.memory,
+		return false, NewInsufficientResourceError(memoryResoureceName, podRequest.memory,
 			getTotalResourceRequest(existingPods).memory, allocatable.Memory().Value())
 	}
-	glog.V(10).Infof("Schedule Pod %+v on Node %+v is allowed, Node is running only %v out of %v Pods.", podName(pod), node, len(pods)-1, allocatable.Pods().Value())
+	glog.V(10).Infof("Schedule Pod %+v on Node %+v is allowed, Node is running only %v out of %v Pods.", podName(pod), nodeName, len(pods)-1, info.Status.Capacity.Pods().Value())
 	return true, nil
 }
 
+// PodFitsResources calculates fit based on requested, rather than used resources
+func (r *ResourceFit) PodFitsResources(pod *api.Pod, existingPods []*api.Pod, node string) (bool, error) {
+	info, err := r.info.GetNodeInfo(node)
+	if err != nil {
+		return false, err
+	}
+	allocatable := info.Status.Allocatable
+	if int64(len(existingPods))+1 > allocatable.Pods().Value() {
```

Comment:
Cool, I thought there was some particular reason that we do not check pod number in Kubelet.